src/test/regress/expected/distributed_domain.out

```diff
@@ -0,0 +1,797 @@
+-- reading violating data of an non-valid constraint errors in citus
+SELECT * FROM use_age_invalid ORDER BY 1;
+ERROR:  value for domain age_invalid violates check constraint "check_age_positive"
+-- should fail since there is data in the table that violates the check
+ALTER DOMAIN age_invalid VALIDATE CONSTRAINT check_age_positive;
+ERROR:  column "b" of table "use_age_invalid_93631045" contains values that violate the new constraint
+CONTEXT:  while executing command on localhost:xxxxx
+DELETE FROM use_age_invalid WHERE b < 0;
+-- should succeed now since the violating data has been removed
+ALTER DOMAIN age_invalid VALIDATE CONSTRAINT check_age_positive;
+-- still fails for constraint
+INSERT INTO use_age_invalid VALUES (5,-1);
+ERROR:  value for domain distributed_domain.age_invalid violates check constraint "check_age_positive"
+CONTEXT:  while executing command on localhost:xxxxx
+SELECT * FROM use_age_invalid ORDER BY 1;
+ a | b
+---------------------------------------------------------------------
+ 1 | 1
+ 2 | 2
+ 3 | 0
+(3 rows)
+
+-- verify we can validate a constraint that is already validated, can happen when we add a node while a domain constraint was not validated
+ALTER DOMAIN age_invalid VALIDATE CONSTRAINT check_age_positive;
+-- test changing the owner of a domain
+SELECT * FROM run_command_on_workers($$ CREATE ROLE domain_owner; $$) ORDER BY 1,2;
+ nodename  | nodeport | success |   result
+---------------------------------------------------------------------
+ localhost |    57637 | t       | CREATE ROLE
+ localhost |    57638 | t       | CREATE ROLE
+(2 rows)
+
+CREATE ROLE domain_owner;
+NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
+HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
+CREATE DOMAIN alter_domain_owner AS int;
+ALTER DOMAIN alter_domain_owner OWNER TO domain_owner;
+SELECT u.rolname
+FROM pg_type t
+         JOIN pg_roles u
+              ON (t.typowner = u.oid)
+WHERE t.oid = 'distributed_domain.alter_domain_owner'::regtype;
+   rolname
+---------------------------------------------------------------------
+ domain_owner
+(1 row)
+
+SELECT * FROM run_command_on_workers($$
+    SELECT u.rolname
+      FROM pg_type t
+      JOIN pg_roles u
+        ON (t.typowner = u.oid)
+     WHERE t.oid = 'distributed_domain.alter_domain_owner'::regtype;
+$$) ORDER BY 1,2;
+ nodename  | nodeport | success |    result
+---------------------------------------------------------------------
+ localhost |    57637 | t       | domain_owner
+ localhost |    57638 | t       | domain_owner
+(2 rows)
+
+DROP DOMAIN alter_domain_owner;
+SET citus.enable_ddl_propagation TO off;
+CREATE DOMAIN alter_domain_owner AS int;
+ALTER DOMAIN alter_domain_owner OWNER TO domain_owner;
+RESET citus.enable_ddl_propagation;
+CREATE TABLE use_alter_domain_owner (a int, b alter_domain_owner);
+SELECT create_distributed_table('use_alter_domain_owner', 'a', shard_count => 4);
```

Comment:
I think I tried that, and would have guessed I kept it.

A bit of a problem with a domain (or a custom type) as a distribution column (in the tests) is that if you drop cascade the schema there are all kind of problems related to the domain being dropped before the distributed table. This gives very strange errors while dropping the table as it can't read the metadata anymore. When deleting a table it reads its shardranges and corresponding hashfunction.

Reply:
The test is still in the file: https://github.com/citusdata/citus/pull/5764/files#diff-4e661126d64871be3ebcb80803d97c56e11c4363f44d9d4716b02306410ddcf3R18-R19

Reply:
> A bit of a problem with a domain (or a custom type) as a distribution column (in the tests) is that if you drop cascade the schema there are all kind of problems related to the domain being dropped before the distributed table. This gives very strange errors while dropping the table as it can't read the metadata anymore. When deleting a table it reads its shardranges and corresponding hashfunction.

That sounds like a bug in how we add distributed tables in the dependency chain. I'd expect postgres to drop things in order of the dependency chain, so first the distributed tables that depend on the domain and only when all those are dropped, the domain.